exchange/xgrid.F90

```diff
@@ -287,17 +287,17 @@ module xgrid_mod
 !> Type to hold pointers for grid boxes
 !> @ingroup xgrid_mod
 type grid_box_type
```

Comment:
This type is public and so are it's fields so we're gonna have to check the component's code to make sure these fields aren't used as pointers anywhere else besides internally (unless we want to make them private now but doubt that'll work). 

This applies to xmap_type as well.

Reply:
Yes. You are right. If component codes are using them, this change certainly breaks something.